Gem/Transfer/ViewsModels/PriceImpactViewModel.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+import Primitives
+import Localization
+
+enum PriceImpactValue: Equatable {
+    case none
+    case low(String)
+    case medium(String)
+    case high(String)
+    case positive(String)
+}
+
+struct PriceImpactViewModel {
+    let fromAssetData: AssetData
+    let toAssetData: AssetData
+    
+    var priceImpact: String { Localized.Swap.priceImpact }
+    
+    private let valueFormatter = ValueFormatter(style: .medium)
+    private var percentFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.locale = .current
+        formatter.numberStyle = .percent
+        formatter.maximumFractionDigits = 2
+        formatter.minimumFractionDigits = 2
+        return formatter
+    }
+    
+    // MARK: - Public methods
+    
+    func priceImpactValue(fromValue: String, toValue: String) -> PriceImpactValue {
+        guard
+            let fromAmount = getSwapAmount(value: fromValue, decimals: fromAssetData.asset.decimals.asInt),
+            let toAmount = getSwapAmount(value: toValue, decimals: toAssetData.asset.decimals.asInt),
+            let fromPrice = getPrice(value: fromAmount, price: fromAssetData.price?.price),
+            let toPrice = getPrice(value: toAmount, price: toAssetData.price?.price)
+        else {
+            return .none
+        }
+        let priceImpact = calculatePriceImpact(fromPrice: fromPrice, expectedPrice: toPrice)
+        let priceImpactValue = evaluatePriceImpactValue(priceImpact: priceImpact)
+        return priceImpactValue
+    }
+    
+    // MARK: - Private methods
+    
+    private func getSwapAmount(value: String, decimals: Int) -> Double? {
+        guard
+            let value = try? valueFormatter.inputNumber(from: value, decimals: decimals),
+            let amount = try? valueFormatter.double(from: value, decimals: decimals)
+        else {
+            return nil
+        }
+        return amount
+    }
+    
+    private func getPrice(value: Double, price: Double?) -> Double? {
```

Comment:
```suggestion
    private func getPrice(value: Double, price: Double?) -> Double? {
```

this is not a price, more like total value? assetFiatValue?